src/main/java/frc/robot/gamepieces/States/ClimberState.java

```diff
@@ -0,0 +1,239 @@
+            downButtonPressed = GamePieceController.getInstance().climberDownButtonPressed;
+
+            climberAL.set(UP);
+            climberAL.setLock(UNLOCK);
+            if (isHighest) {
+                return GameLocked;
+            }
+            if (!upButtonPressed || downButtonPressed) {
+                return GameLocked;
+            }
+
+            return this;
+        }
+
+        public void exit() {
+            // Noop
+        }
+    },
+
+    Retracting {
+        public boolean downButtonPressed;
+
+        public void enter() {
+            // Noop
+        }
+
+        public State action() {
+            downButtonPressed = GamePieceController.getInstance().climberDownButtonPressed;
+
+            climberAL.set(DOWN);
+            climberAL.setLock(UNLOCK);
+            if (!downButtonPressed) {
+                return GameLocked;
+            }
+            if (isLowest) {
+                return GameLocked;
+            }
+            return this;
+        }
+
+        public void exit() {
+            // Noop
+        }
+    },
+
+    GameLocked {
+        public boolean upButtonPressed;
+        public boolean downButtonPressed;
+
+        public void enter() {
+            // Noop
+        }
+
+        public State action() {
+            upButtonPressed = GamePieceController.getInstance().upButtonPressed;
+            downButtonPressed = GamePieceController.getInstance().downButtonPressed;
+
+            climberAL.set(OFF);
+            climberAL.setLock(LOCK);
+            if (upButtonPressed && !downButtonPressed && !isHighest) {
+                return UnlockingUp;
+            }
+            if (downButtonPressed && !upButtonPressed && !isLowest) {
+                return UnlockingDown;
+            }
+            return this;
+        }
+
+        public void exit() {
+            // Noop
+        }
+    },
+
+    Disabled {
+        public void enter() {
+            timer.start();
+        }
+
+        public State action() {
+            climberAL.set(OFF);
+            climberAL.setLock(LOCK);
+            if (timer.get() > disableDelay) { // TODO: determine timer value
+                return GameLocked;
+            }
+            return this;
+        }
+
+        public void exit() {
+            timer.stop();
+            timer.reset();
+        }
+    };
+
+    private static ClimberAL climberAL = ClimberAL.getInstance();
+    private static boolean isLowest = climberAL.isDown();
+    private static boolean isHighest = climberAL.isUp();
+
+    private final static double unlockingDelay = 2.0; // TODO: determine delay in seconds
+    private final static double disableDelay = 3.0; // TODO: determine delay in seconds
+    private final static double climbThreshold = 2.0; // TODO: determine threshold for climber, should this be static or non static
+
+    public static Timer timer = new Timer();
+
+    ClimberState() {
+
+    }
```

Comment:
Normal Java convention is to put variables and constructors at the top of the class definition, not at the bottom.